docs/DeveloperGuide.adoc

```diff
@@ -243,6 +243,52 @@ Step 3. The user executes `activity t/Lunch p/Alex p/David` command to create an
 ** Pros: Easy to implement. Low possibility of bugs.
 ** Cons: Significantly decreases usability of the application.
 
+// tag::expense[]
+=== Expense class
+`Expense` is a class used to record expense information. It contains the following information:
+
+* Paying person
+
+Each expense must be paid by someone, which is recorded as `personId` internally. As the name suggests, only the primary key of the `Person` paying is stored.
+
+* Other involved people
+
+Each expense also involves some people who will be taken to owe money to the paying person. This is recorded as `involvedIds[]` internally, which only stores the primary key of each involved person.
+
+* Amount
+
+The amount of money spent. The value of an `Amount` is stored as a `double`, and ranges from 0.01 to 1000000.
+
+* Description
+
+The description of an expense. The description may be blank if the user chooses to not describe what it is about.
+
+* Deleted flag
+
+This flag tracks whether an expense has been marked as deleted or not, and is stored as a boolean value `isDeleted`. For accountability purposes, we do not allow an `Expense` to be deleted from an `Activity` entirely, and instead use this flag to keep track of it. Expenses can only be fully deleted by deleting the entire activity.
+
+* Settlement flag
+
+SplitWiser supports 2 main types of "expenses": one is an actual expense, and the other is a transaction between people to pay off debts. The latter is what we call a settlement, and is indicated by the boolean `isSettlement`.
```

Comment:
@podocarp @liakify Are we intending to split settlements into a separate class? Otherwise, it is difficult for the UI to distinguish between typical expenses, completed settlements and required transfers.

Reply:
Is the `boolean` insufficient for GUI rendering?

Reply:
The GUI shouldn't be checking the `isSettlement` flag directly - it would violate the separate of concerns between the `Model`/`Logic` (for business logic) and `Ui` (for rendering).